ee_plugin/ui/forms/add_image_collection.py

```diff
@@ -0,0 +1,278 @@
+    )
+
+    percentile_slider = widgets.LabeledSlider(
+        min_value=0,
+        max_value=100,
+        default_value=50,
+        object_name="percentile_value",
+        label="Percentile Value",
+        visible=False,  # Initially hidden
+    )
+
+    composite_group = widgets.build_form_group_box(
+        title="Compositing Options",
+        collapsable=True,
+        collapsed=True,
+        rows=[
+            (
+                QtWidgets.QLabel(
+                    text="Compositing Method",
+                    toolTip="Select how to composite the Image Collection (e.g., Mosaic, Mean, Max, Min, Median, Percentile).",
+                ),
+                composite_combo_box,
+            ),
+        ],
+    )
+
+    def update_percentile_visibility():
+        """Show the percentile slider only when 'Percentile' is selected"""
+        is_percentile = composite_combo_box.currentText() == "Percentile"
+        if (
+            composite_group.findChild(QtWidgets.QWidget, "percentile_value") is None
+            and is_percentile
+        ):
+            composite_group.layout().addRow(percentile_slider)
+
+        percentile_slider.update_label(percentile_slider.slider.value())
+        percentile_slider.set_visibility(is_percentile)
+
+    # Connect signal to update visibility
+    composite_combo_box.currentIndexChanged.connect(update_percentile_visibility)
+
+    dialog = widgets.build_vbox_dialog(
+        windowTitle=_("Add Image Collection"),
+        widgets=[
+            widgets.build_form_group_box(
+                title=_("Source"),
+                rows=[
+                    (
+                        QtWidgets.QLabel(
+                            toolTip=_("The Earth Engine Image Collection ID."),
+                            text="<br />".join(
+                                [
+                                    _("Image Collection ID"),
+                                    "e.g. <code>LANDSAT/LC09/C02/T1_L2</code>",
+                                ]
+                            ),
+                        ),
+                        QtWidgets.QLineEdit(
+                            objectName="image_collection_id",
+                            toolTip=_(
+                                "Enter the ID of the Earth Engine Image Collection (e.g., LANDSAT/LC09/C02/T1_L2)."
+                            ),
+                        ),
+                    ),
+                ],
+            ),
+            widgets.build_form_group_box(
+                title=_("Filter by Properties"),
+                collapsable=True,
+                collapsed=True,
+                rows=[
+                    (
+                        QtWidgets.QLabel(
+                            text=_("Filters:"),
+                            toolTip=_("Add multiple filters by clicking 'Add Filter'."),
+                        ),
+                        widgets.create_filter_widget(),  # Updated: Ensure object name is set
```

Comment:
Not sure what we should do with this comment...  is it worth keeping in the codebase?